scipy/optimize/minpack.py

```diff
@@ -510,28 +523,37 @@ def curve_fit(f, xdata, ydata, p0=None, sigma=None, absolute_sigma=False,
         values will all be 1 (if the number of parameters for the function
         can be determined using introspection, otherwise a ValueError
         is raised).
-    sigma : None or M-length sequence, optional
-        If not None, the uncertainties in the ydata array. These are used as
-        weights in the least-squares problem
-        i.e. minimising ``np.sum( ((f(xdata, *popt) - ydata) / sigma)**2 )``
-        If None, the uncertainties are assumed to be 1.
+    sigma : None or M-length sequence or MxM array, optional
+        Determines the uncertainty in `ydata`. If we define residuals as
```

Comment:
I wouldn't mind to see list bullets here (as it was before). Also I think it can be written better, I'm just not sure how exactly (Like perhaps write `In this case the optimization function is ..."). Maybe you can introduce some improvements.